src/soilbiogeochem/SoilBiogeochemDecompCascadeMIMICSMod.F90

```diff
@@ -0,0 +1,917 @@
+
+               do j = 1,nlev_soildecomp_standard
+                  do fc = 1,num_soilc
+                     c = filter_soilc(fc)
+
+                     if (j==1) o_scalar(c,:) = 0._r8
+
+                     o_scalar(c,1) = o_scalar(c,1) + fr(c,j) * max(o2stress_unsat(c,j), mino2lim)
+                  end do
+               end do
+            else
+               o_scalar(bounds%begc:bounds%endc,1:nlevdecomp) = 1._r8
+            end if
+         else
+            o_scalar(bounds%begc:bounds%endc,1:nlevdecomp) = 1._r8
+         end if
+
+         deallocate(fr)
+
+      else
+
+         ! calculate the rate constant scalar for soil water content.
+         ! Uses the log relationship with water potential given in
+         ! Andren, O., and K. Paustian, 1987. Barley straw decomposition in the field:
+         ! a comparison of models. Ecology, 68(5):1190-1200.
+         ! and supported by data in
+         ! Orchard, V.A., and F.J. Cook, 1983. Relationship between soil respiration
+         ! and soil moisture. Soil Biol. Biochem., 15(4):447-453.
+
+         do j = 1,nlevdecomp
+            do fc = 1,num_soilc
+               c = filter_soilc(fc)
+               psi = min(soilpsi(c,j),maxpsi)
+               ! decomp only if soilpsi is higher than minpsi
+               if (psi > minpsi) then
+                  w_scalar(c,j) = (log(minpsi/psi)/log(minpsi/maxpsi))
+               else
+                  w_scalar(c,j) = 0._r8
+               end if
+            end do
+         end do
+
+         if (use_lch4) then
+            ! Calculate ANOXIA
+            ! Check for anoxia w/o LCH4 now done in controlMod.
+
+            if (anoxia) then
+               do j = 1,nlevdecomp
+                  do fc = 1,num_soilc
+                     c = filter_soilc(fc)
+
+                     o_scalar(c,j) = max(o2stress_unsat(c,j), mino2lim)
+                  end do
+               end do
+            else
+               o_scalar(bounds%begc:bounds%endc,1:nlevdecomp) = 1._r8
+            end if
+         else
+            o_scalar(bounds%begc:bounds%endc,1:nlevdecomp) = 1._r8
+         end if
+
+      end if
+
+      ! Term that reduces decomposition rate at depth
+      do j = 1, nlevdecomp
+         do fc = 1, num_soilc
+            c = filter_soilc(fc)
+            if (use_vertsoilc) then
+               ! Using fixed e-folding depth as in
+               ! SoilBiogeochemDecompCascadeBGCMod.F90
+               depth_scalar(c,j) = exp(-zsoi(j) / decomp_depth_efolding)
+            else
+               depth_scalar(c,j) = 1.0_r8
+            end if
+         end do
+      end do
+
+      ! calculate rates for all litter and som pools
+      ! TODO I will start updating this section when I feel confident in my list
+      !      of k_ factors. For example
+      !      - Should decomp_k be dimensioned (c,j,i_l1m1) instead?
+      !      - Should there be two sets of decomp_k, the first corresponding to
+      !        vmax and the second to km?
```

Comment:
No need for two sets of decomp_k.
In mimics_cycle_CN line 1688 and on, remove the pool (LITm, LITs, ...) from the formulas, and you have the new decomp_k rate.